lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -35,22 +44,79 @@ def collect(cls, block_structure):
         block_structure.request_xblock_fields('is_proctored_enabled')
         block_structure.request_xblock_fields('is_practice_exam')
         block_structure.request_xblock_fields('is_timed_exam')
+        block_structure.request_xblock_fields('entrance_exam_id')
+
+    def transform(self, usage_info, block_structure):
+        """
+        Modify block structure according to the behavior of milestones and special exams.
+        """
+
+        def add_special_exam_info(block_key):
+            """
+            Adds special exam information to course blocks.
+            """
+            if self.is_special_exam(block_key, block_structure):
+
+                #
+                # call into edx_proctoring subsystem
+                # to get relevant proctoring information regarding this
+                # level of the courseware
+                #
+                # This will return None, if (user, course_id, content_id)
+                # is not applicable
+                #
+                timed_exam_attempt_context = None
+                try:
+                    timed_exam_attempt_context = get_attempt_status_summary(
+                        usage_info.user.id,
+                        unicode(block_key.course_key),
+                        unicode(block_key)
+                    )
+                except ProctoredExamNotFoundException as ex:
```

Comment:
This does apply to all special exams, but it is defined and used in the `edx-proctoring` repo, and I don't believe this story should warrant an `edx-proctoring` release.